Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -827,12 +827,26 @@ void ContainerNode::removeChildren()
 
     Ref protectedThis { *this };
     NodeVector removedChildren;
-    removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren);
-
+#if PLATFORM(MAC)
+    if (this != &rootNode() && this->document().settings().opportunisticallyDeleteNodes() && this->document().settings().opportunisticSweepingAndGarbageCollectionEnabled()) {
+        removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren, DeferChildrenChanged::No, CanDelayNodeDeletion::Yes);
```

Comment:
removeAllChildrenWithScriptAssertion is also called in a bunch of other functions in this file.
Why do we not want the same optimization there?